### ISSUE TITLE
Yolov8-segm model training batch removing polygons although they are present in the annotations

### ISSUE THREAD

**Issue by AbhishekBhatti09** [ultralytics/ultralytics#16453]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Train

### Bug

Helllo I have been training a model on detecting pipes from the plumbing plans of a building, so in that while training batch got generated I saw that some of the polygons are getting removed in that training batch although they are present in label files which I have verified by loading those annotations on the annotation tool.

Annotation on the tool :
!

Train Batch of yolov8 :
!

### Environment

Ultralytics YOLOv8.2.19 🚀 Python-3.9.0 torch-2.4.1+cu118 CUDA:0 (NVIDIA TITAN RTX, 24212MiB)
Setup complete ✅ (128 CPUs, 125.7 GB RAM, 919.9/1832.2 GB disk)

### Minimal Reproducible Example

Nothing to add in reproducible I have segmentation annotation of polygons which goes missing for few pipes in yolov8 train batch.

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @AbhishekBhatti09, thank you for reaching out and for your interest in Ultralytics 🚀! This is an automated response to assist you while you wait for a team member to address your issue.

We recommend checking out our [Docs](https://docs.ultralytics.com) for more insights, where you can find useful [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples. Many of the common questions might already be answered there.

Since you are reporting a 🐛 bug, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us diagnose the issue more effectively. This will significantly aid in debugging the problem with your polygon annotations.

If your question relates to custom training ❓, please include detailed information such as dataset image examples and training logs. Ensure that you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Feel free to join and engage with the Ultralytics community:
- For real-time chat, visit [Discord](https://ultralytics.com/discord) 🎧
- For detailed discussions, head over to [Discourse](https://community.ultralytics.com)
- Join the conversation on our [Subreddit](https://reddit.com/r/ultralytics)

## Upgrade

Ensure you are using the latest `ultralytics` package by upgrading all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLOv8 can be run in several verified environments with all dependencies like [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/), and [PyTorch](https://pytorch.org/) preinstalled:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** VM: See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** AMI: See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker**: See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are passing. CI tests validate correct operation on macOS, Windows, and Ubuntu every 24 hours and on each commit.

An Ultralytics engineer will assist you shortly. Thank you for your patience! 😊

**Comment by Y-T-G** [comment]
They will be removed if they are too thin, less than 2 pixels in width or height. Or if the width is over 100 times larger than the height or vice versa.

https://github.com/ultralytics/ultralytics/blob/ba949830bfd232dccd3aa531f6e1d92583ad4dbd/ultralytics/data/augment.py#L1275-L1280

**Comment by AbhishekBhatti09** [comment]
Hello there,

Thanks for your reply but for first my height of instance is 8 pixels for all the instances and about the width 100 times larger than height, I have many such instance where its still there which are bigger than this one got missed. So I dont think that has removed it. because if that would be the case why others bigger than this or equivalent didnt got removed.



> They will be removed if they are too thin, less than 2 pixels in width or height. Or if the width is over 100 times larger than the height or vice versa.
> 
> https://github.com/ultralytics/ultralytics/blob/ba949830bfd232dccd3aa531f6e1d92583ad4dbd/ultralytics/data/augment.py#L1275-L1280

**Comment by Y-T-G** [comment]
From the images you shared, the box that got removed is thinner than the rest.

It could be 8 pixels in your original image, but the model is training on 640x640 images, so it will resize the original image to that size, which means it will become smaller than that.

**Comment by AbhishekBhatti09** [comment]
Hello @Y-T-G ,

I'm already resizing the image to 1280 using my custom letterbox and then manipulating that polygon to have 8 px fix height for all and then giving it to model although I tried the change in annotation by reducing the width of the instance and it worked great so problem was as you stated my ar_thr is greater than probably say 100 default value set in that function ,so now I will just make a change in the code reference you provided in ultralytics to have it limit increased to cover such instance sizes.

Thank you for your answer.

> From the images you shared, the box that got removed is thinner than the rest.
> 
> It could be 8 pixels in your original image, but the model is training on 640x640 images, so it will resize the original image to that size, which means it will become smaller than that.
